Makefile

```diff
@@ -64,6 +65,41 @@ clean: ## Clears cached; deletes node_modules and dist directories
 	rm -rf dist
 	rm -rf node_modules
 
+e2e: | node_modules setup-e2e test-e2e clean-e2e
```

Comment:
Order is not guaranteed here. If we want these to run in order, you need to have them depend on each other. (e2e -> clean -> test -> setup) They should also be PHONY targets.

Reply:
Now added those to PHONY. Please take a look again, thanks!

Reply:
Order is still not guaranteed even with PHONY. You need to make the jobs dependents of each other.

Reply:
Thanks for the offline DM and guidance on how to make the order guaranteed.  
Initially, I separated task per command so that I could use `clean-e2e` independently and reuse it at the same time. However, since it's on the last step of `e2e`, it will require a dependency so that it will guarantee the order which is not I would expect if what I wanted is to clean up but not to run end-to-end testing.
With that, I'm combining all the task into one command to run `e2e` and a separate `clean-e2e` command to clean it up.  There are repetition but I think it's fine.